lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -458,15 +493,26 @@ class QcFlagService {
             for (const newestFlagToBeUpdatedIndex in flagsCreatedBeforeRemovedFlag) {
                 const updatingFlagsCreatedBefore = flagsCreatedBeforeRemovedFlag.slice(Number(newestFlagToBeUpdatedIndex) + 1);
                 for (const updatingFlag of [...updatingFlagsCreatedBefore, ...flagsCreatedAfterRemovedFlag]) {
+                    const { from: updatingFrom, to: updatingTo } = updatingFlag;
                     const effectivePeriods = await QcFlagEffectivePeriodRepository.findAll({
-                        where: {
-                            from: { [Op.lt]: updatingFlag.to },
-                            to: { [Op.gt]: updatingFlag.from },
-                        },
+                        [Op.and]: [
+                            {
+                                [Op.or]: [
+                                    { from: null },
+                                    updatingTo ? { from: { [Op.lt]: updatingTo } } : { from: { [Op.not]: null } },
+                                ],
+                            },
+                            {
+                                [Op.or]: [
+                                    { to: null },
+                                    updatingFrom ? { to: { [Op.gt]: updatingFrom } } : { to: { [Op.not]: null } },
+                                ],
+                            },
+                        ],
```

Comment:
Same thing as before.
However, here you are duplicating code, this should be a dedicated function in the repository.

Reply:
applied